### PR TITLE
Remove sign up functionality

### LINKED ISSUE #499: Restrict account creation to invitation by organization admin
implementation of this probably needs a bit more thinking, but at a minimum:

- [x] remove the "sign up" link in the navbar
- [x] remove or rework the `registrations_controller.rb`
- [x] change the UI when a user accesses a login-only page to notify them that they need to be invited to access it

we probably need some approved text like "You need a POD aggregator account to access this page. Contact an administrator from an Ivy Plus Libraries Consortium member organization for more information."

see also discussion on #475 which led to this implementation issue.

### PR DESCRIPTION
**Pull request by marlo-longley**
Fixes #499  - thanks to @thatbudakguy for pairing on this!

Notes:
- @ggeisler it would be helpful to get your approval of the new flash message text we wrote
- We are going to deploy this branch to staging to make sure we didn't break organization email invites, which uses the same Devise model we just modified for this PR.

### CODE REVIEW

**Comment by cbeer** [comment]
Do we need a rake task or something to create an initial user for local development setup?

**Comment by ggeisler** [comment]
@marlo-longley Could you make a screenshot so I can review the flash message in context?

**Comment by thatbudakguy** [comment]
@cbeer does the `agg:create_user` command in the readme no longer work? idk why it's not a rake task, tbh

**Comment by cbeer** [comment]
@thatbudakguy I guess I should have looked in the README

**Comment by ggeisler** [comment]
@marlo-longley Okay, thanks. Sure, this looks good.

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,11 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
-  devise_for :users, controllers: { invitations: 'organization_invitations', registrations: 'registrations' }
+  # Disable default Devise user registration ("Sign Up"), but support editing user profile when logged in, which is controlled by Devise's RegistrationsController
+  devise_for :users, controllers: { invitations: 'organization_invitations' }, :skip => [:registrations]
+    as :user do
+      get 'users/edit' => 'devise/registrations#edit', :as => 'edit_user_registration'
+      put 'users' => 'devise/registrations#update', :as => 'user_registration'
```
Comment (corylown) [inline config/routes.rb:7]: I think this is preferred ruby style (and consistent with style in the rest of the routes file)
```suggestion
      get 'users/edit' => 'devise/registrations#edit', as: 'edit_user_registration'
      put 'users' => 'devise/registrations#update', as: 'user_registration'
```

Comment (marlo-longley) [reply config/routes.rb:7]: @corylown thanks for the close review!

**Review by corylown — APPROVED** [review]

**Pull request merged**